zcash_client_backend/src/decrypt.rs

```diff
@@ -15,6 +16,17 @@ use zcash_primitives::{
 
 use crate::keys::UnifiedFullViewingKey;
 
+/// An enumeration of the possible relationships a TXO can have to the wallet.
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum TransferType {
+    /// The transfer was received on one of the wallet's external addresses.
+    Incoming,
+    /// The transfer was received on one of the wallet's internal-only addresses.
+    WalletInternal,
+    /// The transfer was decrypted using one of the wallet's outgoing viewing keys.
+    Outgoing,
```

Comment:
Either these cases need more documentation, or the enum as a whole needs more documentation. In particular, it is _almost never_ true that a wallet will detect an outgoing transaction by decrypting with an OVK; instead a change output will be decrypted with an IVK. There are potential edge cases here as well when a transaction is e.g. sending funds between two accounts in a wallet, or involves an explicit send to a wallet's external address as well as a wallet-external address, which cannot be represented as a single instance of this enum.